### PR TITLE
Unifies packaging for plugins

### PR DESCRIPTION
**Pull request by jimczi**
Rename bin/plugins in bin/elasticsearch-plugin

Change the inner structure of the plugins zip:
- The zip must contain a top-level folder titled `elasticsearch` and only one sub-folder titled after the name of the plugin. If you use the gradle build, this structure is automatically generated.

```
<plugin_name>-<version>.zip
|____elasticsearch
| |____<plugin_name>
| | |____<plugin_files, plugin-descriptor.properties>
```

### CODE REVIEW

**Comment by jimferenczi** [comment]
@rjernst this one is for you ;)

File: core/src/main/java/org/elasticsearch/plugins/PluginManager.java [hunk 1]
```diff
@@ -237,7 +239,7 @@ private void extract(PluginHandle pluginHandle, Terminal terminal, Path pluginFi
         Path root = tmp.resolve(pluginHandle.name);
         unzipPlugin(pluginFile, root);
 
-        // find the actual root (in case its unzipped with extra directory wrapping)
+        // find the actual root for the elasticsearch plugin (the zip may contain other system plugins).
         root = findPluginRoot(root);
```
Comment (rmuir) [inline core/src/main/java/org/elasticsearch/plugins/PluginManager.java]: We should not make these changes until we remove site plugin support. This part right here, does not play well with this change at all.

Comment (jimferenczi) [inline core/src/main/java/org/elasticsearch/plugins/PluginManager.java]: @rmuir is it ? I am not sure I understand what you mean by "not play well with this change at all". The change is minimal and should not impact anything. There is a new directory at the root of the plugin (named elasticsearch) that is checked in the PluginManager but the directory will not be present in the plugins directory.

**Comment by jimferenczi** [comment]
@rjernst I've updated this PR in order to reflect the design we all agreed on, can you take a look ?

**Comment by jimferenczi** [comment]
This PR is stalled until @rjernst finishes the work on removing leniency from the plugin cli.

**Comment by jimferenczi** [comment]
Split this PR in two (one for the renaming and one for the repackaging):
https://github.com/elastic/elasticsearch/pull/16453
https://github.com/elastic/elasticsearch/pull/16454

**Pull request closed without merging**